app/models/concerns/wrangleable.rb

```diff
@@ -0,0 +1,23 @@
+module Wrangleable
+  extend ActiveSupport::Concern
+
+  included do
+    after_save :update_last_wrangling_activity, if: :update_wrangling_activity?
+    after_destroy :update_last_wrangling_activity, if: :update_wrangling_activity?
+  end
+
+  private
+
+  def update_last_wrangling_activity
+    User.update_wrangling_activity = false
```

Comment:
I don't think this is the right place to clear `update_wrangling_activity`. If it gets set to `true`, but this hook doesn't run (e.g. if one of the validations fails in `TagsController#update`), it won't get cleared. That means it'll stay true for all subsequent requests involving the same Unicorn worker, at least until it gets cleared by a later, successful save. Possibly one by a different user that wasn't supposed to count as wrangling activity.

IMHO, it'd be better to write a controller method like this:
```ruby
def record_wrangling_activity
  User.update_wrangling_activity = true
  yield
ensure
  User.update_wrangling_activity = false
end
```
If you add that and call `around_action :record_wrangling_activity` for any controller/action combos that should be recorded, you guarantee that `User.update_wrangling_activity` will be set for any actions you want to record, and unset the rest of the time. Using `ensure` means that it'll run even if there are errors, so there's no danger of leaving `User.update_wrangling_activity` set when it's not supposed to be.

Reply:
Yeah, that makes sense! I think I read about that kind of filter in the docs and... promptly forgot all about i :facepalm: Mòran taing!